lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -544,6 +544,62 @@ static LogicalResult verify(spirv::AccessChainOp accessChainOp) {
   return success();
 }
 
+namespace {
+
+// Combine chained `spirv::AccessChainOp` operations into one
+// `spirv::AccessChainOp` operation. The current approach is based on analyzing
+// only two operations at once: the current `spirv::AccessChainOp` and an
+// operation immediately following the current operation.
+struct AccessChainOpCanonicalizer
```

Comment:
Can we call this as `CombineChainedAccessChain` to be clear? The existing name is too general and indicates we have all canonicalization patterns inside. Rather, we'd like to create another class for another canonicalization transformation.

Also related, can we rename `SelectionOpCanonicalizer` to `ConvertSelectionToSelect`?